nexus-and-alo/nexus-strawman.md

```diff
@@ -0,0 +1,361 @@
+        if err != nil {
+            log.Fatal(err)
+        }
+        defer client.Close()
+
+        // Build worker with Temporal pieces
+        temporalClient, err := temporalclient.Dial(temporalclient.Options{HostPort: "nexus-backend.example.com:7233"})
+        if err != nil {
+            log.Fatal(err)
+        }
+        defer temporalClient.Close()
+        builder, err := temporalnexus.NewWorkerBuilder(temporalnexus.WorkerBuilderOptions{Client: temporalClient})
+        if err != nil {
+            log.Fatal(err)
+        }
+        // Add workflow as ALO to Nexus
+        err = builder.AddWorkflowALOHandler("foo", temporalnexus.WorkflowALOHandlerOptions{
+            Workflow: FooWorkflow,
+            Options:  temporalclient.StartWorkflowOptions{TaskQueue: "my-task-queue"},
+        })
+        if err != nil {
+            log.Fatal(err)
+        }
+
+        // Worker definition...
+    }
+    ```
+1. Nexus is now able to process requests to its frontend endpoint.
+
+### Consuming a Nexus Service from Temporal
+
+1. User asks the Nexus Service author for the endpoint of the Nexus Service Instance they are trying to consume. For Temporal based Nexus Service Instances, the endpoint will not be HTTP or gRPC but rather a custom uri formulation.
+2. User registers provided endpoint with their SDK to avoid hardcoding the endpoint into all of their application logic. 
+3. User calls the external Nexus Service using a Workflow-specific Nexus API which is provided by Temporal.
+    ```go
+    // calling a Nexus from a Temporal Workflow
+    func MyWorkflow(ctx workflow.Context, name string) (string, error) {
+        client := temporalnexus.NewAloClient(temporalnexus.AloClientOptions{Endpoint: "temporal:72aA64A0jhOL99B2"})
+        // Start some ALO
+        alo, err := client.StartAlo(ctx, temporalnexus.StartAloOptions{
+            Name: "foo",
+            ID:   "my-id",
+            Arg:  name,
+        })
+        if err != nil {
+            return "", err
+        }
+        // Workflow implementation...
+    }
+    ```
+
+## Concepts
+
+### Nexus Instantiation
+
+Instantiating a Nexus Service makes it possible for the Nexus Backend to receive traffic meant for that specific Instance. Nexus Instantiation does make your Nexus Service Instance endpoint(s) available for traffic, but until you register and run corresponding Nexus Task Workers, no forward progress will be made on incoming requests. In this way, sending requests at this stage is similar to starting Workflows without any Workflow Workers running.
+
+**Registration:**
+
+For the sake of this proposal, we will only consider the case where Nexus is registered via `nexus` CLI. In the future we may evaluate and consider alternative methods of registering a Nexus Service Instance. Registering a Nexus Service Instance would be done with the following command structure:
+
+`nexus service register --name my-service --description "My service" --backend nexus-backend.example.com` 
```

Comment:
I thought a service has a type and an instance name, is this no longer the case?

Reply:
I've removed the concept of type entirely from the proposal. I do believe it will come back but its a two way door.